WhenWorkingWithAPetMap.java

```diff
@@ -0,0 +1,99 @@
+package serenitylabs.tutorials.vetclinic.collections.exercises;
+
+import org.junit.Test;
+import serenitylabs.tutorials.vetclinic.Breed;
+import serenitylabs.tutorials.vetclinic.Pet;
+
+import java.util.*;
+
+import static org.hamcrest.Matchers.*;
+import static org.junit.Assert.assertThat;
+
+public class WhenWorkingWithAPetMap {
+
+    @Test
+    public void pets_can_be_found_by_their_name() {
+        Map<String, Pet> pets = new HashMap<>();
+
+        Pet fido = Pet.dog().named("Fido");
+
+        pets.put("Fido", fido);
+
+        Set<String> keysOfSet = pets.keySet();
+
+        assertThat(fido,equalTo(pets.get("Fido")));
+    }
+
+    @Test
+    public void should_be_able_to_get_a_default_value_if_no_matching_key_is_present() {
+        Map<String, Pet> pets = new HashMap<>();
+
+        Pet fido = Pet.dog().named("Fido");
+        Pet stray = Pet.dog().named("Stray");
+
+        pets.put("Fido", fido);
+        pets.put("Stray", stray);
+
+        String defaultValue = null;
+        assertThat(defaultValue, equalTo(pets.get("Megha")));
```

Comment:
A map will return null if there is no value present. The idea of a default value is to return a specific non-null value when there is no matching element. Here, you have the defaultValue set to null, so that is not what we want.

In Java 8, we use the getOrDefault() method to do this, e.g.:

```
        Map<String, Pet> pets = new HashMap<>();

        Pet fido = Pet.dog().named("Fido");
        Pet stray = Pet.dog().named("Stray");

        pets.put("Fido", fido);
        pets.put("Stray", stray);

        Pet retrievedPet = pets.getOrDefault("Rover", stray);

        assertThat(retrievedPet, equalTo(stray));
```
This will return `stray` whenever there is no matching key in the map.